src/Agent.Worker/Release/Artifacts/GitHubClient.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Net.Http;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+using Newtonsoft.Json;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release.Artifacts
+{
+    [ServiceLocator(Default = typeof(GitHubClient))]
+    public interface IGitHubClient : IAgentService
+    {
+        GitHubRepository GetUserRepo(string accessToken, string repository);
+    }
+
+    public class GitHubClient : AgentService, IGitHubClient
+    {
+        private const string GithubRepoUrlFormat = "https://api.github.com/repos/{0}";
+
+        public GitHubRepository GetUserRepo(string accessToken, string repositoryName)
+        {
+            string errorMessage;
+            string url = StringUtil.Format(GithubRepoUrlFormat, repositoryName);
+            var repository = QueryItem<GitHubRepository>(accessToken, url, out errorMessage);
+
+            if (errorMessage != null)
+            {
+                throw new InvalidOperationException(errorMessage);
+            }
+
+            return repository;
+        }
+
+        private static T QueryItem<T>(string accessToken, string url, out string errorMessage)
```

Comment:
why Private Static?
I don't think you need 3 separate method for sending a http request to github, just put all logic into GetUserRepo(), since it's not huge.